src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -266,74 +252,184 @@ public static bool IsDefaultEqualsOperator(IMethodBodyOperation operation)
                 assignedValues.SetEquals(fields);
         }
 
-        public static (ImmutableArray<ArgumentSyntax> initializerArguments, ImmutableArray<SyntaxNode> assignmentsToRemove)
-            GetInitializerValuesForNonPrimaryConstructor(
+        public static (ImmutableArray<ArgumentSyntax> initializerArguments,
+            ImmutableArray<SyntaxNode> assignmentsToRemove)
+        GetInitializerValuesForNonPrimaryConstructor(
             IConstructorBodyOperation operation,
             ImmutableArray<IPropertySymbol> positionalParams)
         {
-            var body = GetBlockOfMethodBody(operation);
-            var bodyOps = body?.Operations ?? ImmutableArray<IOperation>.Empty;
             var _1 = ArrayBuilder<ArgumentSyntax>.GetInstance(out var initializerBuilder);
             var _2 = ArrayBuilder<SyntaxNode>.GetInstance(out var removalBuilder);
 
             // make sure the assignment wouldn't reference local variables we may have declared
-            var assignmentValues = GetAssignmentValuesForConstructor(bodyOps,
+            var assignmentValues = GetAssignmentValuesForConstructor(operation,
                 assignment => IsSafeAssignment(assignment)
                     ? assignment.Syntax as ExpressionSyntax
                     : null);
 
-            foreach (var property in positionalParams)
-            {
-                var assignmentIndex = assignmentValues.IndexOf(value => property.Equals(value.left));
-                if (assignmentIndex == -1)
+            AddArgumentAndRemovalAssignments(assignmentValues, positionalParams, initializerBuilder, removalBuilder);
+
+            return (initializerBuilder.AsImmutable(), removalBuilder.AsImmutable());
+        }
+
+        public static (ImmutableArray<ArgumentSyntax> initializerArguments,
+            ImmutableArray<AssignmentExpressionSyntax> assignmentsToRemove)
```

Comment:
this signature is really difficult to read.

Reply:
for all of these it woudl be good to have docs saying what the purpose of hte method is and maybe some example of what these do given an input.